### ISSUE TITLE
First epoch is extremely slow

### ISSUE THREAD

**Issue by mrT23** [NVIDIA/DALI#1415]
i am following the example from
https://github.com/NVIDIA/DeepLearningExamples/tree/master/PyTorch/Classification/RN50v1.5

my first epoch is extremely slow, i am doing an imagenet epoch in 20-30 minutes on 8Vx100 server.
from the second epoch and forth, i am getting much better speeds, less than 2 minutes per epoch 

is it a bug or normal behavior?
can we do a faster "warmup" for dali loader?

thanks

**Comment by JanuszL** [comment]
Hi,
Please check your CPU, GPU, and disk IO utilization during the first epoch.
I guess it is because you are using some slow storage and during the first epoch, data is [cached in RAM ](https://blog.webnersolutions.com/understanding-disk-cache-in-linux), in the following one data is read directly from RAM and this is fast.
What I would recommend is to use some fast SSD raid storage. DALI will help when CPU is a bottleneck, but now then you cannot read your training data from the drive fast enough.

**Comment by mrT23** [comment]
@JanuszL , Thanks for the answer
your intuition is right, it is a problem of reading from storage in the first epoch.
after upgrading the storage, first epoch is faster (also still slower than the second one)

is this caching mechanism a Linux auto caching, or Nividia Dali directly caches the data ?

thanks

**Comment by JanuszL** [comment]
This caching is strictly a Linux thing. DALI doesn't do such thing except for some special case you [can read about here](https://docs.nvidia.com/deeplearning/sdk/dali-developer-guide/docs/supported_ops.html#nvidia.dali.ops.ImageDecoder) - we can cache data at the decoder stage in GPU memory but it is rather for the selected use case (many GPUs, shard size that can fit into GPU memory).
